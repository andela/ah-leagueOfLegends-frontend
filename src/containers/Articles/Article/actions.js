import axios from 'axios';

import { BACKEND_URL } from '../../../utils/config';
import { FETCH_ARTICLE, FETCH_SUCCESS, FETCH_FAILURE } from './constant';

export const fetchArticle = () => ({ type: FETCH_ARTICLE });

// eslint-disable-next-line
const articleFectSuccsess = payload => ({ type: FETCH_SUCCESS, payload: payload.data.article.article });

const articleFectFailure = error => ({ type: FETCH_FAILURE, error });

const fetchArticles = slug => (dispatch) => {
  dispatch(fetchArticle());
  axios.get(`${BACKEND_URL}api/articles/${slug}`)
    .then((res) => {
      dispatch(articleFectSuccsess(res));
    })
    .catch(err => dispatch(articleFectFailure(err)));
};

export default fetchArticles;
