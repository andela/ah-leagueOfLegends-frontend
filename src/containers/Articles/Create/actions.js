import axios from 'axios';

import { BACKEND_URL } from '../../../utils/config';
import { ARTICLE_SUCCESS, ARTICLE_FAILURE, PUBLISH_ARTICLE } from './constants';

export const articleSuccess = payload => ({ type: ARTICLE_SUCCESS, payload });

export const articleFailure = errors => ({ type: ARTICLE_FAILURE, errors });

export const articleFetch = () => ({ type: PUBLISH_ARTICLE });

const publishArticle = (data, history, update = false, slug) => (dispatch) => {
  dispatch(articleFetch());
  let MAIN_URL = `${BACKEND_URL}api/articles`;
  if (update) {
    MAIN_URL = `${BACKEND_URL}api/articles/${slug}`;
  }
  const token = localStorage.getItem('access_token');
  // eslint-disable-next-line
  axios.defaults.headers.common = { Authorization: 'Bearer ' + token }
  if (update) {
    axios.put(MAIN_URL, data)
      .then((res) => {
        dispatch(articleSuccess(res));
        history.push(`/article/${res.data.articles.slug}`);
      })
      .catch(err => dispatch(articleFailure(err)));
  }
  axios.post(MAIN_URL, data)
    .then((res) => {
      dispatch(articleSuccess(res));
      history.push(`/article/${res.data.articles.slug}`);
    })
    .catch(err => dispatch(articleFailure(err)));
};

export default publishArticle;
