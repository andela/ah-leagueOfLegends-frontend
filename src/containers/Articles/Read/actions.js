import axios from 'axios';

import { BACKEND_URL } from '../../../utils/config';
import { ARTICLE_SUCCESS, ARTICLE_FAILURE, GET_ARTICLES } from './constants';

export const articleSuccess = payload => ({ type: ARTICLE_SUCCESS, payload });

export const articleFailure = errors => ({ type: ARTICLE_FAILURE, errors });

export const articleFetch = () => ({ type: GET_ARTICLES });

export const getArticles = (limit, offset) => (dispatch) => {
  dispatch(articleFetch());
  const MAIN_URL = ` ${BACKEND_URL}api/articles?limit=${limit}&offset=${offset}`;
  axios.get(MAIN_URL)
    .then(res => dispatch(articleSuccess(res)))
    .catch(err => dispatch(articleFailure(err)));
};
