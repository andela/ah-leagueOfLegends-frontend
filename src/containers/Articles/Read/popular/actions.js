import axios from 'axios';

import { BACKEND_URL } from '../../../../utils/config';
import { ARTICLE_SUCCESS, ARTICLE_FAILURE, GET_ARTICLES } from '../constants';

export const articleSuccess = payload => ({ type: ARTICLE_SUCCESS, payload });

export const articleFailure = errors => ({ type: ARTICLE_FAILURE, errors });

export const articleFetch = () => ({ type: GET_ARTICLES });

export const allArticles = () => (dispatch) => {
  dispatch(articleFetch());
  const MAIN_URL = ` ${BACKEND_URL}api/articles`;
  axios.get(MAIN_URL)
    .then(res => dispatch(articleSuccess(res)))
    .catch(err => dispatch(articleFailure(err)));
};
