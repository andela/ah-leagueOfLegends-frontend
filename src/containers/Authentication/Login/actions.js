import LoginConstants from './actiontypes';
import history from '../../../history';

const axios = require('axios');

const loginRequest = () => ({ type: LoginConstants.LOGIN_REQUEST });
const loginSuccess = payload => ({
  type: LoginConstants.LOGIN_SUCCESS,
  payload,
});
const loginFailure = error => ({
  type: LoginConstants.LOGIN_FAILURE,
  error,
});
function login(email, password) {
  return (dispatch) => {
    dispatch(loginRequest);
    return axios.post('https://ah-leagueoflegends-staging.herokuapp.com/api/users/login/',
      { user: { email, password } },
      { headers: { 'Content-Type': 'application/json' } })
      .then((response) => {
        const elem = document.querySelectorAll('.modal')[0];
        const elem1 = document.querySelectorAll('.modal-overlay')[0];
        dispatch(loginSuccess(response.data));
        const { token } = response.data.user;
        localStorage.setItem('access_token', token);
        history.push('/');
        elem.style = 'display: none';
        elem1.style = 'display: none';
      })
      .catch((error) => {
        dispatch(loginFailure(error.response.data.errors.error[0]));
      });
  };
}
export default login;
