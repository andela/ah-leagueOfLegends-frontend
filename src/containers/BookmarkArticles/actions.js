import axios from 'axios';
import { BOOKMARK_ARTICLE, BOOKMARK_ARTICLE_FAILURE } from './ActionTypes';
import { BACKEND_URL } from '../../utils/config';
import fetchArticles from '../Articles/Article/actions';

export const bookmarkArticle = () => ({ type: BOOKMARK_ARTICLE });

export const bookmarkArticleFailure = errors => ({
  type: BOOKMARK_ARTICLE_FAILURE,
  errors,
});

// Bookmark an article
export const bookmarkArticles = (slug) => {
  const token = localStorage.getItem('access_token');
  axios.defaults.headers.common.Authorization = `Bearer ${token}`;
  const url = `${BACKEND_URL}api/articles/${slug}/bookmark/`;
  return (dispatch) => {
    dispatch(bookmarkArticle());
    return axios.put(url)
      .then(
        () => {
          dispatch(fetchArticles(slug));
        },
        (errors) => {
          dispatch(bookmarkArticleFailure(errors));
        },
      );
  };
};
