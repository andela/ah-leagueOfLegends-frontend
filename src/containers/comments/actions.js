import axios from 'axios';

import { BACKEND_URL } from '../../utils/config';
import { COMMENT_SUCCESS, COMMENT_FAILURE, COMMENT_FETCH } from './constants';

export const commentSuccess = payload => ({ type: COMMENT_SUCCESS, payload });

export const commentFailure = error => ({ type: COMMENT_FAILURE, error });

export const commentFetch = () => ({ type: COMMENT_FETCH });


const publishComment = (payloadData, slug, method, update = false, commentId) => (dispatch) => {
  let data = { comment: { body: payloadData } };
  if (method === 'PUT') {
    data = { body: payloadData };
  }
  dispatch(commentFetch());
  let MAIN_URL = `${BACKEND_URL}api/articles/${slug}/comments`;
  const token = localStorage.getItem('access_token');
  if (update) {
    MAIN_URL = `${BACKEND_URL}api/articles/${slug}/comments/${commentId}`;
  }
  axios.defaults.headers.common = { Authorization: `Bearer ${token}` };
  switch (method) {
    case 'POST':
      axios.post(MAIN_URL, data)
        .then((res) => {
          dispatch(commentSuccess(res));
        })
        .catch(err => dispatch(commentFailure(err)));
      break;
    case 'PUT':
      axios.put(MAIN_URL, data, slug, commentId)
        .then((res) => {
          dispatch(commentSuccess(res));
        })
        .catch(err => dispatch(commentFailure(err)));
      break;
    case 'DELETE':
      axios.delete(MAIN_URL)
        .then((res) => {
          dispatch(commentSuccess(res));
        })
        .catch(err => dispatch(commentFailure(err)));
      break;
    default:
      axios.get(MAIN_URL)
        .then((res) => {
          dispatch(commentSuccess(res));
        })
        .catch(err => dispatch(commentFailure(err)));
  }
};

export default publishComment;
