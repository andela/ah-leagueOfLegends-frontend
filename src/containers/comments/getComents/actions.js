import axios from 'axios';

import { FETCH_COMMENT, FETCH_COMMENT_FAIL, FETCH_COMMENT_SUCCSESS } from './constant';
import { BACKEND_URL } from '../../../utils/config';


export const commentSuccess = payload => ({ type: FETCH_COMMENT_SUCCSESS, payload });

export const commentFailure = error => ({ type: FETCH_COMMENT_FAIL, error });

export const commentFetch = () => ({ type: FETCH_COMMENT });

export const getComments = slug => (dispatch) => {
  const MAIN_URL = `${BACKEND_URL}api/articles/${slug}/comments`;
  axios.get(MAIN_URL)
    .then(res => dispatch(commentSuccess(res)))
    .catch(err => dispatch(commentFailure(err)));
};
