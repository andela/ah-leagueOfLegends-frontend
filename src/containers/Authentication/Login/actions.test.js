import configureStore from 'redux-mock-store';

import * as LoginActions from './actions';

const mockStore = configureStore();
const store = mockStore();

describe('loginActions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  describe('loginSuccess', () => {
    test('dispatches the correct action and payload', () => {
      const user = {
        user: {
          email: 'testuser@gmail.com',
          password: 'letmein',
        },
      };
      const expectedActions = [
        {
          type: 'USERS_LOGIN_SUCCESS',
          payload: user,
        },
      ];
      store.dispatch(LoginActions.loginSuccess(user));
      expect(store.getActions()).toEqual(expectedActions);
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('loginFailure', () => {
    test('dispatches the correct action and payload', () => {
      const error = 'User with that email and password was not found';
      const expectedActions = [
        {
          type: 'USERS_LOGIN_FAILURE',
          error,
        },
      ];
      store.dispatch(LoginActions.loginFailure(error));
      expect(store.getActions()).toEqual(expectedActions);
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
