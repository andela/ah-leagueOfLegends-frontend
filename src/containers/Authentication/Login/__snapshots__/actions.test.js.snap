// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginActions loginFailure dispatches the correct action and payload 1`] = `
Array [
  Object {
    "error": "User with that email and password was not found",
    "type": "USERS_LOGIN_FAILURE",
  },
]
`;

exports[`loginActions loginSuccess dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "user": Object {
        "email": "testuser@gmail.com",
        "password": "letmein",
      },
    },
    "type": "USERS_LOGIN_SUCCESS",
  },
]
`;
